testing/test_collection.py

```diff
@@ -647,7 +647,7 @@ def test_pkgfile(self, testdir):
         col = testdir.getnode(config, x)
         assert isinstance(col, pytest.Module)
         assert col.name == "x.py"
-        assert col.parent.parent is None
+        assert col.parent.parent.parent is None
```

Comment:
lets please add add omment here to explain the nesting since it is a tad confusing as of now

Reply:
Agreed. Probably checking the `name` of each parent will also help clarifying things. 👍

Reply:
It's pretty simple, it's basically testing to see that the Session's parent is None, which should always be and the nodes have been correctly nested. For a more visual example:

```
tmpdir
 \- subdir
     \- __init__.py
     \- x.py


Session<>
   ^
   | (parent)
Package<subdir>
   ^
   | (parent)
Module<x>
```